modules/citrus-integration/src/it/java/com/consol/citrus/javadsl/runner/GlobalVariablesTestRunnerIT.java

```diff
@@ -0,0 +1,95 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.consol.citrus.javadsl.runner;
+
+import com.consol.citrus.annotations.CitrusResource;
+import com.consol.citrus.annotations.CitrusTest;
+import com.consol.citrus.context.TestContext;
+import com.consol.citrus.dsl.runner.TestRunner;
+import com.consol.citrus.dsl.testng.TestNGCitrusTestRunner;
+import org.testng.annotations.Optional;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+
+/**
+ * Tests the loading and usage of global variables in citrus
+ */
+@Test
+public class GlobalVariablesTestRunnerIT extends TestNGCitrusTestRunner {
+
+    @CitrusTest
+    public void globalProperties() {
+        echo("Project name is: ${project.name}");
+
+        echo("Testing global variables from properties: ${globalWelcomingText}");
+    }
+
+    @CitrusTest
+    @Parameters({"runner", "context"})
+    public void randomLocalVariable_readMultipleTimes_valueShouldRemainConstantOnEachRead(
+            @Optional @CitrusResource TestRunner runner, @Optional @CitrusResource TestContext context) {
+
+        runner.variable("localRandomNumber1", "citrus:randomNumber(5)");
+        runner.echo("localRandomNumber1: ${localRandomNumber1}");
+        assertVariableValuesMatchWhenReadMultipleTimes(context, "localRandomNumber1");
+
+        runner.createVariable("localRandomNumber2", "citrus:randomNumber(8)");
+        runner.echo("localRandomNumber2: ${localRandomNumber2}");
+        assertVariableValuesMatchWhenReadMultipleTimes(context, "localRandomNumber2");
+    }
+
+    @CitrusTest
+    @Parameters({"runner", "context"})
+    public void randomGlobalVariable_readMultipleTimes_valueShouldRemainConstantOnEachRead(
+            @Optional @CitrusResource TestRunner runner, @Optional @CitrusResource TestContext context) {
+        runner.echo("globalRandomNumber1: ${globalRandomNumber1}");
+        assertVariableValuesMatchWhenReadMultipleTimes(context, "globalRandomNumber1");
+    }
+
+    @CitrusTest
+    @Parameters({"runner", "context"})
+    public void multipleGlobalVariablesWithCrossReferencesAndFunctions_readMultipleTimes_valueShouldRemainConstantOnEachRead(
+            @Optional @CitrusResource TestRunner runner, @Optional @CitrusResource TestContext context) {
+        runner.echo("globalSum1: ${globalSum1}");
+        assertVariableValuesMatchWhenReadMultipleTimes(context, "globalSum1");
+
+        runner.echo("globalSum2: ${globalSum2}");
+        assertVariableValuesMatchWhenReadMultipleTimes(context, "globalSum2");
+
+        assertVariableValuesMatch(context, "globalSum1", "globalSum2");
+    }
+
+    private void assertVariableValuesMatchWhenReadMultipleTimes(final TestContext testContext, final String name) {
+        String val1 = testContext.getVariable(name);
```

Comment:
I think this method does not test what you want it to test.
`testContext.getVariable(name);` gets you the unresolved value of the variable. No dynamic replacement of functions or variables has been applied here.

Even if I rollback your changes and run the tests, they pass.
Here you can see why:
![image](https://user-images.githubusercontent.com/25754312/50592970-ba00ab00-0e96-11e9-80c6-3ab336145fc7.png)
`val1` and `val2` both contain `citrus:randomNumber(5)` what makes them equal.

```suggestion
        String val1 = testContext.resolveDynamicValue(testContext.getVariable(name));
```
If you use this also for `val2`, the test fails as expected before you applied your change.
![image](https://user-images.githubusercontent.com/25754312/50593135-617ddd80-0e97-11e9-91bd-26ce0a6de51a.png)

When I now add back your changes to the code, the tests pass as they should.

Reply:
Well spotted. I assumed that when using the citrus java dsl a call to getVariable would return the value with any dynamic content resolved. I incorporated your suggested change. 
On a side note - what is the recommended way to access variables using the java DSL in citrus? Calling resolveDynamicValue is not intuitive, at least it wasn't for me ;-)

Reply:
Well, now that you ask, I think that this is the only way to do it currently. :thinking: 
As a review of the current `TestContext` concept is planned for the next major, this should be addressed then.